temporalcli/commands.operator_search_attribute.go

```diff
@@ -0,0 +1,167 @@
+package temporalcli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/temporalio/cli/temporalcli/internal/printer"
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/operatorservice/v1"
+)
+
+func (c *TemporalOperatorSearchAttributeCreateCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	if len(c.Name) != len(c.Type) {
+		return fmt.Errorf("number of name and type options should be the same")
+	}
+
+	listReq := &operatorservice.ListSearchAttributesRequest{
+		Namespace: c.Parent.Parent.Namespace,
+	}
+	existingSearchAttributes, err := cl.OperatorService().ListSearchAttributes(cctx, listReq)
+	if err != nil {
+		return fmt.Errorf("unable to get existing search attributes: %w", err)
+	}
+
+	searchAttributes := make(map[string]enums.IndexedValueType, len(c.Type))
+	for i, saType := range c.Type {
+		saName := c.Name[i]
+		typeInt, err := stringToEnum(saType)
+		if err != nil {
+			return fmt.Errorf("unable to parse search attribute type %s: %w", saType, err)
+		}
+		existingSearchAttributeType, searchAttributeExists := existingSearchAttributes.CustomAttributes[saName]
+		if searchAttributeExists && existingSearchAttributeType != typeInt {
+			return fmt.Errorf("search attribute %s already exists and has different type %s", saName, existingSearchAttributeType.String())
+		}
+		searchAttributes[saName] = typeInt
+	}
+
+	request := &operatorservice.AddSearchAttributesRequest{
+		SearchAttributes: searchAttributes,
+		Namespace:        c.Parent.Parent.Namespace,
+	}
+
+	_, err = cl.OperatorService().AddSearchAttributes(cctx, request)
+	if err != nil {
+		return fmt.Errorf("unable to add search attributes: %w", err)
+	}
+	cctx.Printer.Println(color.GreenString("Search attributes have been added"))
+	return nil
+}
+
+func stringToEnum(search string) (enums.IndexedValueType, error) {
```

Comment:
If you must handle case insensitive, you can loop over the `enums.IndexedValueType_shorthandValue` and check the key to match this instead of a switch here. But if you do want to keep this helper, can you name it something more specific since it's a package level function visible for all writing code in this package?

Reply:
will use more specific name

Reply:
It's not future proof for future enum values and it's more lines of code than if you just did a simple map iteration, but up to you I guess.

Reply:
updated to use enums.IndexedValueType_shorthandValue